### ISSUE TITLE
Add-on icon can't be changed from DevHub Edit information page

### ISSUE THREAD

**Issue by AlexandraMoga** [mozilla/addons-server#8900]
STR:

1. Log in to Developer Hub
2. Navigate to Manage my submissions and choose one of your add-ons to edit
3. Change the add-on icon to one of the predefined options or upload a custom icon
4. Save changes

Actual result:
The default add-on icon is still displayed after the changes have been saved

Expected result:
The updated icon (predefined or custom) should be displayed after the changes have been saved

Notes:
 - reproduced on AMO -dev with FF61, Win10x64
 - looks like a regression from https://github.com/mozilla/addons-server/issues/5271

![icon not set](https://user-images.githubusercontent.com/31961530/42816409-5d7af0c2-89d3-11e8-9b56-ae465dd7ec39.gif)

**Comment by EnTeQuAk** [comment]
This sounds like a Django 1.11 regression.

I can take a look at this.

**Comment by EnTeQuAk** [comment]
Yeah, this happened when I re-implemented the `IconTypeSelect` widget and I missed the label/input fields which leads to nothing being saved.

**Comment by AlexandraMoga** [comment]
@EnTeQuAk sadly I can still reproduce the issue for the case when a custom icon is uploaded. I'm inclined to re-open the issue...

**Comment by EnTeQuAk** [comment]
I'll look into that.

**Comment by AlexandraMoga** [comment]
Verified on AMO -dev with FF61, Win10x64

Custom add-on icons can now be uploaded from DevHub:

![custom icon](https://user-images.githubusercontent.com/31961530/43144038-db869a2a-8f64-11e8-8642-5ced3ddc051f.gif)

I can see that there is another issue here at icon preview rendering: the 64x64 preview is generated but the 32x32 is displayed only after the page is refreshed. Since this is only a small issue, I will file a bug to track it separately.

**Comment by EnTeQuAk** [comment]
The problem there is that they're generated asynchronously, imho the
same issue exists in prod too. Filing an low priority issue about sounds
good though.

On Tue, Jul 24, 2018, at 4:31 PM, Alexandra Gal-Moga wrote:
> Verified on AMO -dev with FF61, Win10x64


> Custom add-on icons can now be uploaded from DevHub:


> custom icon


> I can see that there is another issue here at icon preview rendering:
> the 64x64 preview is generated but the 32x32 is displayed only after
> the page is refreshed. Since this is only a small issue, I will file a
> bug to track it separately.> — You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub[1], or mute the
> thread[2].
Links:

  1. https://github.com/mozilla/addons-server/issues/8900#issuecomment-407427687
